### DISCUSSION TITLE
Flag --views in make:controller --resource and make:model -a

### DISCUSSION THREAD

**Discussion by PovilasKorop** [laravel/framework#39576] [Ideas]
After my [first PR being merged](https://twitter.com/taylorotwell/status/1458127204043341828), I decided to think what else could make our lives easier related to make:model and make:controller

And, for a typical CRUD scenario, we probably need Views.
I've [raised a Twitter discussion](https://twitter.com/PovilasKorop/status/1458705557926391810), and so far the idea seems to resonate with people.

What developers tend to do is create those views manually, and then write code in controllers to use those views, manually again.
So we could automate the part of creating the files.

Of course, views are very individual, based on custom design, so by default Blade files would be empty.

So:

```
php artisan make:controller PostController --resource --views
```

would:

1. Create a folder `resources/views/posts`
2. Create three **empty** files inside: `index.blade.php`, `create.blade.php` and `edit.blade.php`
3. Modify Controller methods to use those views: just static `return view('posts.index')` and respectively in `create()` and `edit()` methods.
4. [optional with much bigger scope] We could come up with a Stubs system which would use the developer's prepared stubs to use the stub contents in those blade files. Maybe as a separate future PR.

What do you think?

**Comment by entity** [comment]
I think for CRUD related stuff, this would be absolutely fantastic, more so if we could then have integration within Laravel Breeze/Jetstream. I definitely think what you've mentioned about having dev prepared stubs would be superb, even just a simple listing of the models, and a form to add, to give a base to work from.

**Reply by PovilasKorop** [reply]
You misunderstood my idea a bit. At this point, I want to generate EMPTY Blade files, so just the structure, to avoid creating files manually and referencing them in the Controller.

The content/design of Blade files is so so individual that I don't think it would be easy to build something flexible enough.

**Reply by entity** [reply]
Apologies I did understand, I was just referencing your last part that longer term it would be create to have dev created STUBS.

But with this flag, it would mean that when using in say an inertia Vue app, it would create the .Vue version.

if that makes sense

**Reply by PovilasKorop** [reply]
Yes, it totally makes sense. It's just WAY out of scope of what I'm trying to propose here.
The stub idea needs a separate brainstorming session, or maybe it's best to leave that to external packages, there are a lot of full CRUD generators on the market.

**Comment by nticaric** [comment]
Would love to see this in the upcoming versions of laravel!

**Comment by PovilasKorop** [comment]
Update on this one: I decided to not pursue it after all, as it turns out to be a huge amount of work:

- Implement `make:view` command which would be a new structure, different from default `make:xxx` generators as it doesn't generate a PHP class
- Call that `make:view` from `make:controller` and `make:model` commands but only under some conditions
- Fill in the Controller stubs with `return view('xxxxx')` statements, but only under some conditions
- Take care of all possible subfolders, already existing views, etc.
- Write tests for all of those

And this is not even starting on the **content** of the views, any type of stubs or `@extends('xxxxx')` or `<x-app-layout>` is a separate huge amount of work, which is mostly covered now by external packages, so I conclude this is exactly the direction this idea should go.

For generating views, I recommend this old package - https://github.com/svenluijten/artisan-view - or just google "laravel make view" and you will find more alternatives.
